### DISCUSSION TITLE
Trouble installing

### DISCUSSION THREAD

**Discussion by John-Krause** [ARM-DOE/pyart#1464] [General]
I installed anaconda 3.0 and then tried to install pyart with:

 conda install -c conda-forge arm_pyart

I got this error:

Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Solving environment: unsuccessful attempt using repodata from current_repodata.json, retrying with next repodata source.

Can someone suggest a better install method?

**Comment by mgrover1** [comment]
@John-Krause - what version of python do you have installed in your environment?

**Reply by John-Krause** [reply]
john.krause@hops:~$ python --version
Python 3.11.4

**Reply by mgrover1** [reply]
and what platform is this on? Mac, Windows, or Linux? Can you a `conda list` and post the output here?

**Reply by mgrover1** [reply]
I would also recommend updating your anaconda versions, or consider switching to mambaforge distributions
- Mambaforge download (recommended) https://github.com/conda-forge/miniforge#mambaforge
- Anaconda release notes https://docs.anaconda.com/free/anaconda/reference/release-notes/

**Reply by John-Krause** [reply]
eventually (hours later......) the flexible solver figured it out:

Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Solving environment: done

## Package Plan ##

  environment location: /home/john.krause/anaconda3

  added / updated specs:
    - arm_pyart


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge
    _openmp_mutex-4.5          |            2_gnu          23 KB  conda-forge
    anaconda-anon-usage-0.4.2  |  py311hfc0e8ea_0          28 KB
    anaconda-cloud-auth-0.1.3  |     pyhd8ed1ab_0          19 KB  conda-forge
    anaconda-navigator-2.5.0   |  py311h06a4308_0         5.6 MB
    arm_pyart-1.15.1           |  py311h459d7ec_0         7.8 MB  conda-forge
    arrow-cpp-11.0.0           |       h374c478_2        10.2 MB
    aws-c-common-0.6.8         |       h7f98852_0         169 KB  conda-forge
    aws-c-event-stream-0.1.6   |       h6a678d5_6          25 KB
    aws-checksums-0.1.11       |      ha492f5f_12          50 KB  conda-forge
    aws-sdk-cpp-1.8.185        |       h721c034_1         2.2 MB
    ca-certificates-2023.7.22  |       hbcca054_0         146 KB  conda-forge
    certifi-2023.7.22          |     pyhd8ed1ab_0         150 KB  conda-forge
    cftime-1.6.2               |  py311h4c7f6c3_1         242 KB  conda-forge
    conda-23.7.4               |  py311h38be061_0         1.3 MB  conda-forge
    cryptography-41.0.4        |  py311h63ff55d_0         2.0 MB  conda-forge
    curl-8.2.1                 |       hdbd6064_0          83 KB
    fontconfig-2.14.1          |       h4c34cd2_2         281 KB
    grpc-cpp-1.48.2            |       he1ff14a_1         4.8 MB
    h5netcdf-1.2.0             |     pyhd8ed1ab_0          41 KB  conda-forge
    hdf4-4.2.13                |    h10796ff_1005         949 KB  conda-forge
    keyutils-1.6.1             |       h166bdaf_0         115 KB  conda-forge
    krb5-1.20.1                |       h81ceb04_0         1.3 MB  conda-forge
    lat_lon_parser-1.3.0       |     pyhd8ed1ab_0          14 KB  conda-forge
    libarchive-3.6.2           |       h6ac8c49_2         900 KB
    libclang-14.0.6            |default_hc6dbbc7_1         137 KB
    libclang13-14.0.6          |default_he11475f_1         9.8 MB
    libcups-2.4.2              |       h2d74bed_1         4.5 MB
    libcurl-8.2.1              |       h251f7ec_0         399 KB
    libevent-2.1.12            |       hf998b51_1         417 KB  conda-forge
    libgcc-ng-13.2.0           |       h807b86a_2         753 KB  conda-forge
    libgomp-13.2.0             |       h807b86a_2         411 KB  conda-forge
    libmamba-1.4.1             |       h2dafd23_1         1.7 MB
    libmambapy-1.4.1           |  py311h2dafd23_1         289 KB
    libnetcdf-4.8.1            |       h8322cc2_2         1.2 MB
    libnghttp2-1.52.0          |       h2d74bed_1         672 KB
    libpq-12.15                |       hdbd6064_1         2.4 MB
    libssh2-1.10.0             |       ha35d2d1_2         233 KB  conda-forge
    libstdcxx-ng-13.2.0        |       h7e041cc_2         3.7 MB  conda-forge
    libthrift-0.15.0           |       h1795dd8_2         4.0 MB
    libxkbcommon-1.0.3         |       he3ba5ed_0         581 KB  conda-forge
    libxml2-2.10.4             |       hcbfbd50_0         755 KB
    libxslt-1.1.37             |       h873f0b0_0         251 KB  conda-forge
    libzip-1.8.0               |       h1c5bbd1_1         126 KB  conda-forge
    lxml-4.9.3                 |  py311hdbbb534_0         1.5 MB
    mysql-5.7.20               |    hf484d3e_1001        79.2 MB  conda-forge
    navigator-updater-0.4.0    |  py311h06a4308_1         2.3 MB
    netcdf4-1.6.2              |  py311h0e679e6_0         453 KB
    numba-0.57.1               |  py311h96b013e_0         5.3 MB  conda-forge
    open-radar-data-0.0.5      |     pyhd8ed1ab_0          12 KB  conda-forge
    openssl-3.1.3              |       hd590300_0         2.5 MB  conda-forge
    pkce-1.0.3                 |     pyhd8ed1ab_0           9 KB  conda-forge
    pycurl-7.45.1              |  py311hae980a4_3          76 KB  conda-forge
    pydantic-1.10.12           |  py311h459d7ec_1         2.2 MB  conda-forge
    python-3.11.5              |       h955ad1f_0        32.7 MB
    python-dotenv-1.0.0        |     pyhd8ed1ab_1          23 KB  conda-forge
    python_abi-3.11            |          2_cp311           5 KB  conda-forge
    qt-main-5.15.2             |       h7358343_9        53.7 MB
    qt-webengine-5.15.9        |       h9ab4d14_7        53.8 MB
    qtwebkit-5.212             |       h3fafdc1_5        16.2 MB
    setuptools-scm-7.1.0       |     pyhd8ed1ab_0          38 KB  conda-forge
    setuptools_scm-7.1.0       |       hd8ed1ab_0           6 KB  conda-forge
    tokenizers-0.13.3          |  py311h1b04a43_0         3.9 MB  conda-forge
    tomli-2.0.1                |     pyhd8ed1ab_0          16 KB  conda-forge
    versioneer-0.29            |     pyhd8ed1ab_0          47 KB  conda-forge
    xarray-datatree-0.0.12     |     pyhd8ed1ab_0          54 KB  conda-forge
    xmltodict-0.13.0           |     pyhd8ed1ab_0          13 KB  conda-forge
    xradar-0.0.13              |     pyhd8ed1ab_0          60 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       324.3 MB

The following NEW packages will be INSTALLED:

  anaconda-anon-usa~ pkgs/main/linux-64::anaconda-anon-usage-0.4.2-py311hfc0e8ea_0 
  anaconda-cloud-au~ conda-forge/noarch::anaconda-cloud-auth-0.1.3-pyhd8ed1ab_0 
  arm_pyart          conda-forge/linux-64::arm_pyart-1.15.1-py311h459d7ec_0 
  cftime             conda-forge/linux-64::cftime-1.6.2-py311h4c7f6c3_1 
  h5netcdf           conda-forge/noarch::h5netcdf-1.2.0-pyhd8ed1ab_0 
  hdf4               conda-forge/linux-64::hdf4-4.2.13-h10796ff_1005 
  keyutils           conda-forge/linux-64::keyutils-1.6.1-h166bdaf_0 
  lat_lon_parser     conda-forge/noarch::lat_lon_parser-1.3.0-pyhd8ed1ab_0 
  libclang13         pkgs/main/linux-64::libclang13-14.0.6-default_he11475f_1 
  libcups            pkgs/main/linux-64::libcups-2.4.2-h2d74bed_1 
  libnetcdf          pkgs/main/linux-64::libnetcdf-4.8.1-h8322cc2_2 
  libzip             conda-forge/linux-64::libzip-1.8.0-h1c5bbd1_1 
  mysql              conda-forge/linux-64::mysql-5.7.20-hf484d3e_1001 
  netcdf4            pkgs/main/linux-64::netcdf4-1.6.2-py311h0e679e6_0 
  open-radar-data    conda-forge/noarch::open-radar-data-0.0.5-pyhd8ed1ab_0 
  pkce               conda-forge/noarch::pkce-1.0.3-pyhd8ed1ab_0 
  pydantic           conda-forge/linux-64::pydantic-1.10.12-py311h459d7ec_1 
  python-dotenv      conda-forge/noarch::python-dotenv-1.0.0-pyhd8ed1ab_1 
  python_abi         conda-forge/linux-64::python_abi-3.11-2_cp311 
  setuptools-scm     conda-forge/noarch::setuptools-scm-7.1.0-pyhd8ed1ab_0 
  setuptools_scm     conda-forge/noarch::setuptools_scm-7.1.0-hd8ed1ab_0 
  tomli              conda-forge/noarch::tomli-2.0.1-pyhd8ed1ab_0 
  versioneer         conda-forge/noarch::versioneer-0.29-pyhd8ed1ab_0 
  xarray-datatree    conda-forge/noarch::xarray-datatree-0.0.12-pyhd8ed1ab_0 
  xmltodict          conda-forge/noarch::xmltodict-0.13.0-pyhd8ed1ab_0 
  xradar             conda-forge/noarch::xradar-0.0.13-pyhd8ed1ab_0 

The following packages will be UPDATED:

  anaconda-navigator                  2.4.2-py311h06a4308_0 --> 2.5.0-py311h06a4308_0 
  arrow-cpp                          11.0.0-py311ha0d4687_0 --> 11.0.0-h374c478_2 
  aws-c-common       pkgs/main::aws-c-common-0.4.57-he6710~ --> conda-forge::aws-c-common-0.6.8-h7f98852_0 
  aws-c-event-stream                       0.1.6-h2531618_5 --> 0.1.6-h6a678d5_6 
  aws-checksums      pkgs/main::aws-checksums-0.1.9-he6710~ --> conda-forge::aws-checksums-0.1.11-ha492f5f_12 
  aws-sdk-cpp                            1.8.185-hce553d0_0 --> 1.8.185-h721c034_1 
  cryptography       pkgs/main::cryptography-41.0.2-py311h~ --> conda-forge::cryptography-41.0.4-py311h63ff55d_0 
  curl                                     8.1.1-h37d81fd_2 --> 8.2.1-hdbd6064_0 
  fontconfig                              2.14.1-h52c9d5c_1 --> 2.14.1-h4c34cd2_2 
  grpc-cpp                                1.46.1-h33aed49_1 --> 1.48.2-he1ff14a_1 
  libarchive                               3.6.2-hab531cd_0 --> 3.6.2-h6ac8c49_2 
  libclang                        10.0.1-default_hb85057a_2 --> 14.0.6-default_hc6dbbc7_1 
  libcurl                                  8.1.1-h91b91d3_2 --> 8.2.1-h251f7ec_0 
  libevent            pkgs/main::libevent-2.1.12-h8f2d780_0 --> conda-forge::libevent-2.1.12-hf998b51_1 
  libgcc-ng          pkgs/main::libgcc-ng-11.2.0-h1234567_1 --> conda-forge::libgcc-ng-13.2.0-h807b86a_2 
  libgomp              pkgs/main::libgomp-11.2.0-h1234567_1 --> conda-forge::libgomp-13.2.0-h807b86a_2 
  libmamba                                 1.4.1-hdb19cb5_0 --> 1.4.1-h2dafd23_1 
  libmambapy                          1.4.1-py311hdb19cb5_0 --> 1.4.1-py311h2dafd23_1 
  libstdcxx-ng       pkgs/main::libstdcxx-ng-11.2.0-h12345~ --> conda-forge::libstdcxx-ng-13.2.0-h7e041cc_2 
  libxkbcommon       pkgs/main::libxkbcommon-1.0.1-hfa300c~ --> conda-forge::libxkbcommon-1.0.3-he3ba5ed_0 
  libxml2                                 2.9.14-h74e7548_0 --> 2.10.4-hcbfbd50_0 
  libxslt              pkgs/main::libxslt-1.1.35-h4e12654_0 --> conda-forge::libxslt-1.1.37-h873f0b0_0 
  lxml                                4.9.1-py311h1edc446_0 --> 4.9.3-py311hdbbb534_0 
  navigator-updater                   0.4.0-py311h06a4308_0 --> 0.4.0-py311h06a4308_1 
  numba              pkgs/main::numba-0.57.0-py311ha02d727~ --> conda-forge::numba-0.57.1-py311h96b013e_0 
  openssl              pkgs/main::openssl-1.1.1w-h7f8727e_0 --> conda-forge::openssl-3.1.3-hd590300_0 
  python                                  3.11.4-h7a1cb2a_0 --> 3.11.5-h955ad1f_0 
  qt-main                                 5.15.2-h327a75a_7 --> 5.15.2-h7358343_9 
  qt-webengine                            5.15.9-hd2b0992_4 --> 5.15.9-h9ab4d14_7 
  qtwebkit                                 5.212-h4eab89a_4 --> 5.212-h3fafdc1_5 
  tokenizers         pkgs/main::tokenizers-0.13.2-py311he7~ --> conda-forge::tokenizers-0.13.3-py311h1b04a43_0 

The following packages will be SUPERSEDED by a higher-priority channel:

  _libgcc_mutex           pkgs/main::_libgcc_mutex-0.1-main --> conda-forge::_libgcc_mutex-0.1-conda_forge 
  _openmp_mutex          pkgs/main::_openmp_mutex-5.1-1_gnu --> conda-forge::_openmp_mutex-4.5-2_gnu 
  ca-certificates    pkgs/main::ca-certificates-2023.08.22~ --> conda-forge::ca-certificates-2023.7.22-hbcca054_0 
  certifi            pkgs/main/linux-64::certifi-2023.7.22~ --> conda-forge/noarch::certifi-2023.7.22-pyhd8ed1ab_0 
  conda              pkgs/main::conda-23.7.4-py311h06a4308~ --> conda-forge::conda-23.7.4-py311h38be061_0 
  krb5                    pkgs/main::krb5-1.20.1-h568e23c_1 --> conda-forge::krb5-1.20.1-h81ceb04_0 
  libssh2              pkgs/main::libssh2-1.10.0-h37d81fd_2 --> conda-forge::libssh2-1.10.0-ha35d2d1_2 
  pycurl             pkgs/main::pycurl-7.45.2-py311h37d81f~ --> conda-forge::pycurl-7.45.1-py311hae980a4_3 

The following packages will be DOWNGRADED:

  libnghttp2                              1.52.0-ha637b67_1 --> 1.52.0-h2d74bed_1 
  libpq                                    12.15-h37d81fd_1 --> 12.15-hdbd6064_1 
  libthrift                               0.15.0-h0d84882_2 --> 0.15.0-h1795dd8_2 


Proceed ([y]/n)?

**Reply by mgrover1** [reply]
Yeah, I suspect migrating to mamba here would help! You can install it using

```bash
conda install mamba
```
then
```bash
mamba install arm_pyart
```
or the package of your choosing. It utilizes parallel solving + a C++ codebase.

**Comment by zssherman** [comment]
I second the mamba install as well. I've also run into the solver stall with older environments as well, depending on the environment etc, might need a new environment as well.

**Comment by mgrover1** [comment]
@John-Krause - did any of those suggestions work?

**Reply by John-Krause** [reply]
eventually (hours later......) the flexible solver figured it out:

I did not attempt any other installation methods

**Reply by mgrover1** [reply]
Sounds good - I am glad it finally resolved
